plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitManager.java

```diff
@@ -324,6 +331,15 @@ private Iterator<HivePartitionMetadata> getPartitionMetadata(
 
         Iterator<List<HivePartition>> partitionNameBatches = partitionExponentially(hivePartitions, minPartitionBatchSize, maxPartitionBatchSize);
         Iterator<List<HivePartitionMetadata>> partitionBatches = transform(partitionNameBatches, partitionBatch -> {
+            // Use dynamic filters to reduce the partitions listed by getPartitionsByNames
+            TupleDomain<ColumnHandle> currentDynamicFilter = dynamicFilter.getCurrentPredicate();
+            if (!currentDynamicFilter.isAll()) {
+                TupleDomain<ColumnHandle> partitionsFilter = currentDynamicFilter.intersect(tableHandle.getCompactEffectivePredicate());
+                partitionBatch = partitionBatch.stream()
+                        .filter(hivePartition -> partitionMatches(tableHandle.getPartitionColumns(), partitionsFilter, hivePartition))
```

Comment:
Do you have benchmark results?

Reply:
![Screenshot 2024-10-07 at 12 03 24 PM](https://github.com/user-attachments/assets/55f3a787-717e-4922-8f4d-b36f13b21100)
[Hive DF partition metadata pruning.pdf](https://github.com/user-attachments/files/17277045/Hive.DF.partition.metadata.pruning.pdf)

Reply:
I don't understand the results. Do they mean caching helps the most?

Reply:
It means that this optimization helps when the metadata is not already in the cache. Catching of metadata is of course much better. We are not worse with metdata pruning + caching, there's just some duration flakiness on a few queries.